lib/matplotlib/tests/test_text.py

```diff
@@ -919,6 +919,7 @@ def call(*args, **kwargs):
 
 
 def test_metrics_cache2():
+    plt.close('all')
```

Comment:
All tests should start with no figure open due to the test fixture, so this should be unnecessary?

Reply:
This was a long shot that something was getting not cleared because the assert that is failing is before we do anything in this test.